pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +89,54 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func Retry(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
```

Comment:
Worth some function comments to describe the semantics of this call. I'd be inclined to leave it called Wait as well, given we're not retrying on any errors, only the single case where there's an in-flight collision.